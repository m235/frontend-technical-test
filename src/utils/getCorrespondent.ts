import type { Conversation } from '@/types/conversation'

export const getCorrespondent = (userId: number, conversation: Conversation) => {
  switch (userId) {
    case conversation.senderId:
      return conversation.recipientNickname
    case conversation.recipientId:
      return conversation.senderNickname
    default:
      throw new Error('You are nor the recipient nor the sender')
  }
}
