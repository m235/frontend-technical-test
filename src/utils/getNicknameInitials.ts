export const getNicknameInitials = (nickname: string) =>
  nickname
    .split(' ')
    .map((string) => string.trim().charAt(0))
    .slice(0, 2)
    .join('')
