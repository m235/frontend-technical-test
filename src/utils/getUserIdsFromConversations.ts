import type { Conversation } from '@/types/conversation'

export const getUserIdsFromConversations = (conversations?: Conversation[]) =>
  Array.from<number>(new Set(conversations?.reduce((acc, curr) => [...acc, curr.recipientId, curr.senderId], []) ?? []))
