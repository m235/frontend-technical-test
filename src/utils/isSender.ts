import type { Conversation } from '@/types/conversation'

export const isSender = (userId: number, conversation: Conversation) => userId === conversation.senderId
