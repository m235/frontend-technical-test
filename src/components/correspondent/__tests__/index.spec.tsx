import useAuth from '@/hooks/useAuth'
import { render } from '@/tests/utils'
import Correspondent from '@/components/correspondent'

const mockedUseAuth = useAuth as jest.Mock<any>
jest.mock('@/hooks/useAuth')

describe('components:correspondent', () => {
  const conversation = { id: 1, senderNickname: 'John Travolta', senderId: 1, recipientNickname: 'Henri', recipientId: 2 }

  beforeEach(() => {
    mockedUseAuth.mockImplementation(() => ({ userId: 1, user: [{ id: 1, nickname: 'John Travolta', token: 'x' }] }))
  })
  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should render nothing', async () => {
    mockedUseAuth.mockImplementation(() => ({ userId: null }))

    const { container } = render(<Correspondent />)

    expect(container).toMatchSnapshot()
    expect(container.childElementCount).toEqual(0)
  })
  it('should render correctly', async () => {
    const { container, getByText } = render(<Correspondent conversation={conversation} />)

    expect(container).toMatchSnapshot()
    expect(getByText('Henri')).toBeTruthy()
  })
})
