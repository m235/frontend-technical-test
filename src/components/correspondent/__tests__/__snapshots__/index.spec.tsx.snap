// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:correspondent should render correctly 1`] = `
<div>
  <span>
    Henri
  </span>
</div>
`;

exports[`components:correspondent should render nothing 1`] = `<div />`;
